bitcoin-magazine/bitcoin-explained/explaining-bitcoin-addresses.md

```diff
@@ -0,0 +1,649 @@
+
+Aaron Van Wirdum: 00:09:27
+
+hang on hang on because we're not actually using fingers most of the time so I want to make this clear that it just means there we have a decimal system so that means we have 10 different symbols that represent a number that's right with this symbol 0 which is a round thing and
+
+Sjors Provoost: 00:09:45
+
+this probably not a coincidence that that happens to match.
+
+Aaron Van Wirdum: 00:09:48
+
+I totally agree, I just want to make it clear that we're not actually using fingers most of the time.
+
+Sjors Provoost: 00:09:53
+
+No, and so the...
+
+Aaron Van Wirdum: 00:09:54
+
+Okay, so we have 10 symbols, so that means that once you get by the 11th number, at that point you're going to have to reuse symbols you've already used, so you're now going to use combinations. So in our case, that would be, well it's going to get confusing because the first number is a zero, so then the 11th number is the one and the zero.
+
+Sjors Provoost: 00:10:13
+
+Yeah exactly, and there have been different bases in use, right? So base 360 I believe was used by Babylonians, or maybe base 60. And then for computers we tend to use base 2 internally, because chips are either on or off, so it's 0 or 1, so a long series of 0s and 1s. And you can express any number of that now in order to read machine code typically you would use hexadecimal which is base 16 so that is 0 to 9 and then a to F mm-hmm yeah and right and so base 58 is basically this 58 possible characters to express something with.
+
+Aaron Van Wirdum: 00:10:51
+
+Yeah, it's all numbers and there's different ways of expressing a number based on your base. That determines how many symbols you're using.
+
+Sjors Provoost: 00:11:02
+
+Right, the trade-off here is readability really, because you could represent machine code as normal characters, so the ASCII alphabet, or the ASCII character set is 256 different characters, So that's base 256. But if you've ever done something like print and then the name of a file, your computer will show complete gibberish on the screen and it will start beeping. And the reason it starts beeping is because one of these codes Somewhere in the base 256 is a beep, which actually makes your terminal beep. So it is completely impractical to view a file using base 256, even though there is a character for every of the 256 things there. So that's why you tend to do that in base 16. Hexadecimal is relatively easy to read, but then it's quite long. If you take a public key and you write it as hexadecimal it's a rather long thing to write down but and bass 58 is a little bit shorter so maybe you know it's easier to copy paste perhaps or it's not even easy to read on the phone like bass 58 is pretty terrible because it's uppercase, lowercase, uppercase, lowercase.
```

Comment:
```suggestion
Right, the trade-off here is readability really, because you could represent machine code as normal characters, so the ASCII alphabet, or the ASCII character set is 256 different characters, So that's base 256. But if you've ever done something like print and then the name of a file, your computer will show complete gibberish on the screen and it will start beeping. And the reason it starts beeping is because one of these codes Somewhere in the base 256 is a beep, which actually makes your terminal beep. So it is completely impractical to view a file using base 256, even though there is a character for every of the 256 things there. So that's why you tend to do that in base 16. Hexadecimal is relatively easy to read, but then it's quite long. If you take a public key and you write it as hexadecimal it's a rather long thing to write down but and base 58 is a little bit shorter so maybe you know it's easier to copy paste perhaps or it's not even easy to read on the phone like base 58 is pretty terrible because it's uppercase, lowercase, uppercase, lowercase.
```